src/ApiCode.ts

```diff
@@ -1,48 +1,65 @@
-async function stopIDToFirst5Buses(stopID : string): Promise<JSON[]> {
+async function stopIDToFirst5Buses(stopID : string): Promise<Bus[]> {
     const  myUrl : string = "https://api.tfl.gov.uk/StopPoint/" + stopID + "/Arrivals";
     const response = await fetch(myUrl);
     const data = await response.json();
     // deal with JSON response
     const busesArray = []
     const numOfBuses = Math.min(5, data.length)
     for (let i = 0 ; i < numOfBuses;  i ++){
-        busesArray.push(data[i])
+        busesArray.push({lineName : data[i].lineName, timeToStation: data[i].timeToStation})
     }
     return busesArray
 
 
   }
 
+  export interface Bus {
+    lineName: string
+    timeToStation: string
+  }
+
+
+  export interface BusStop {
+   
+    stationName: string;
+    buses : Bus[]
+  
+}
+
 async function postCodeToLongLat(postcode : string) : Promise<string[]>  {
     const myUrl : string = "https://api.postcodes.io/postcodes/" + postcode ;
     const response = await fetch(myUrl);
     const data = await response.json();
     return [data.result.longitude , data.result.latitude];
 }
 
-async function longLatToBusStopIds(longLat : string[]) : Promise<string[]> {
+async function longLatToBusStopIds(longLat : string[]) : Promise<string[][]> {
     const  myUrl : string = "https://api.tfl.gov.uk/StopPoint?stopTypes=NaptanPublicBusCoachTram&lat=" + longLat[1] + "&lon=" + longLat[0];
     const response = await fetch(myUrl);
     const data = await response.json();
     // deal with JSON response
     const busIds = []
+    const busStopNames = []
     const numOfStops = Math.min(2, data.stopPoints.length)
     for (let i = 0 ; i < numOfStops;  i ++){
         busIds.push(data.stopPoints[i].naptanId);
+        busStopNames.push(data.stopPoints[i].commonName + " (" + data.stopPoints[i].indicator + ")")
     }
-    return busIds
+    return [busIds, busStopNames]
```

Comment:
I think it would perhaps be simpler and make more sense to have a single `busStops` array, where each `busStop` contains properties for both the ID and name.